inst/webshot.js

```diff
@@ -23,60 +23,57 @@ var opt_defaults = {
 // =====================================================================
 var args = system.args;
 
-if (args.length < 3) {
+if (args.length < 2) {
   console.log('Usage:\n' +
-    '  phantomjs webshot.js <url> <name>.png [options]');
+    '  phantomjs webshot.js <data>');
   phantom.exit(1);
 }
 
-var url = args[1];
-var filename = args[2];
-var opts = utils.parseArgs(args.slice(3));
-opts = utils.merge(opt_defaults, opts);
-
-// These should be numbers
-if (opts.vwidth)  opts.vwidth  = +opts.vwidth;
-if (opts.vheight) opts.vheight = +opts.vheight;
-if (opts.delay)   opts.delay   = +opts.delay;
-if (opts.zoom)    opts.zoom    = +opts.zoom;
-
-// This should be four numbers separated by ","
-if (opts.cliprect) {
-  opts.cliprect = opts.cliprect.split(",");
-  opts.cliprect = opts.cliprect.map(function(x) { return +x; });
-}
-
-// Can be 1 or 4 numbers separated by ","
-if (opts.expand) {
-  opts.expand = opts.expand.split(",");
-  opts.expand = opts.expand.map(function(x) { return +x; });
-  if (opts.expand.length !== 1 && opts.expand.length !== 4) {
-    console.log("'expand' must have either 1 or 4 values.");
-    phantom.exit(1);
-  }
-}
-
-// Can have multiple selectors
-if (opts.selector) {
-  opts.selector = opts.selector.split(",");
-}
+var data = JSON.parse(args[1]);
 
 // =====================================================================
 // Screenshot
 // =====================================================================
-casper.start(url).zoom(opts.zoom)
-  .viewport(opts.zoom * opts.vwidth, opts.zoom * opts.vheight);
 
-if (opts.delay > 0)
-  casper.wait(opts.delay * 1000);
+casper.start();
 
-if (opts.eval) {
-  eval(opts.eval);
-}
+casper.eachThen(data, function(response) {
+  var url = response.data.url;
+  var file = response.data.file;
+  var opts = response.data;
+
+  // Prepare options
+  opts = utils.merge(opt_defaults, opts);
```

Comment:
That has different behavior: it will overwite values in `opts` with those from `opt_defaults`. I think it would make sense to add a (shallow) object cloning function called `utils.clone`.